src/aooptionsdialog.cpp

```diff
@@ -216,8 +216,37 @@ AOOptionsDialog::AOOptionsDialog(QWidget *parent, AOApplication *p_ao_app)
 
   ui_log_timestamp_cb = new QCheckBox(ui_form_layout_widget);
 
+  connect(ui_log_timestamp_cb, SIGNAL(stateChanged(int)), this, SLOT(timestamp_cb_changed(int)));
+
   ui_gameplay_form->setWidget(row, QFormLayout::FieldRole, ui_log_timestamp_cb);
 
+  row += 1;
+  ui_log_timestamp_format_lbl = new QLabel(ui_form_layout_widget);
+  ui_log_timestamp_format_lbl->setText(tr("Log timestamp format:\n") + QDateTime::currentDateTime().toString(ao_app->get_log_timestamp_format()));
+  ui_gameplay_form->setWidget(row, QFormLayout::LabelRole, ui_log_timestamp_format_lbl);
+
+  ui_log_timestamp_format_combobox = new QComboBox(ui_form_layout_widget);
+  ui_log_timestamp_format_combobox->setEditable(true);
+
+  QString l_current_format = ao_app->get_log_timestamp_format();
+
+  ui_log_timestamp_format_combobox->addItem(l_current_format);
+  if (l_current_format != "h:mm:ss AP")
+    ui_log_timestamp_format_combobox->addItem("h:mm:ss AP"); // 2:13:09 PM
+  if (l_current_format != "hh:mm:ss")
+    ui_log_timestamp_format_combobox->addItem("hh:mm:ss"); // 14:13:09
+  if (l_current_format != "h:mm AP")
+    ui_log_timestamp_format_combobox->addItem("h:mm AP"); // 2:13 PM
+  if (l_current_format != "hh:mm")
+    ui_log_timestamp_format_combobox->addItem("hh:mm"); // 14:13
```

Comment:
I don't see the point of these if statements. Instead of adding l_current_format as an item, set the current value of the widget to it.

Reply:
i may be stupid